### PR TITLE
End multiplexer stream once one of its inputs end.

### PR DESCRIPTION
**Pull request by eskimor**
Also add test, that we don't panic once a stream is exhausted.

### CODE REVIEW

**Review by ordian — APPROVED** [review]

File: node/network/bridge/src/multiplexer.rs [hunk 1]
```diff
@@ -105,6 +110,17 @@ impl Stream for RequestMultiplexer {
 	}
 }
 
+impl FusedStream for RequestMultiplexer {
```
Comment (ordian) [inline node/network/bridge/src/multiplexer.rs:113]: can we remove this `.fuse()` now?
https://github.com/paritytech/polkadot/blob/4866f8e72029e719d7ce74e341bb1e94ebd724f7/node/network/bridge/src/lib.rs#L172

**Comment by ordian** [comment]
An alternative would be what @drahnr suggested, it seems even simpler:
https://github.com/paritytech/polkadot/pull/2352#discussion_r570061531

**Comment by eskimor** [comment]
> An alternative would be what @drahnr suggested, it seems even simpler:
> [#2352 (comment)](https://github.com/paritytech/polkadot/pull/2352#discussion_r570061531)

Actually yes, don't know why I picked the harder way. Should not really matter either way.

**Comment by eskimor** [comment]
Ok strictly speaking this PR alters logic, but it is small only touches new code, got reviewed and fixes a panic - so quick merge should be fine.

**Comment by eskimor** [comment]
bot merge

**Pull request merged**